### ISSUE TITLE
Loss is not decreasing with eml loss function

### ISSUE THREAD

**Issue by Adrian398** [SenJia/EML-NET-Saliency#4]
First of all, thank you very much for your great work and repository. I tried to retrain your model with salicon data. While training the training_loss is not decreasing. As you can see here:
![train_loss_eml](https://user-images.githubusercontent.com/28985882/85429700-6947ab00-b57f-11ea-85ee-cfa6152149b5.png)

When I am using the mse everything works fine:
![train_loss_mse](https://user-images.githubusercontent.com/28985882/85429745-79f82100-b57f-11ea-9a03-4faba2dc4af7.png)

I also printed all of the loss values:
<img width="596" alt="nss_biggest" src="https://user-images.githubusercontent.com/28985882/85429848-998f4980-b57f-11ea-9abc-c50d9823d33a.png">

As you can see NSS is 25 and so up to 50x larger than the other two could this be the reason the network can't train?

**Comment by SenJia** [comment]
Hi Adrian398, I am glad you like my study. I suppose the X-axis represents the number of training epochs, and you didn't combine NSS when applying MSE. The losses could be at different scales, but the point is that the combined should(in theory) approach zero when there is no error, imagine the losses of anchor location and semantic label for object detection. Could you please check the output of your model and compute any metrics on the validation set, I guess the output may look reasonable. Otherwise, could you reduce the initial learning rate by 1e-1, let me know if you still cannot obtain a reasonable prediction.
